rust-symcrypt/src/rsa/mod.rs

```diff
@@ -0,0 +1,716 @@
+            // Currently, only two prime RSA is supported, i.e. the only valid indexes are 0 and 1
+            let prime_1 = symcrypt_sys::SymCryptRsakeySizeofPrime(self.inner.0, 0);
+            let prime_2 = symcrypt_sys::SymCryptRsakeySizeofPrime(self.inner.0, 1);
+            (prime_1, prime_2)
+        }
+    }
+
+    /// `export_key_pair_blob()` returns a [`RsaKeyPairExportBlob`] value.
+    pub fn export_key_pair_blob(&self) -> Result<RsaKeyPairExportBlob, SymCryptError> {
+        // Get size of primes only once
+        let (size_p, size_q) = self.get_size_of_primes();
+
+        // Allocate buffers for filling RsaKeyPairExportBlob
+        let mut modulus_buffer = vec![0u8; self.get_size_of_modulus() as usize];
+        let mut pub_exp = vec![0u64; 1];
+        let mut p = vec![0u8; size_p as usize];
+        let mut q = vec![0u8; size_q as usize];
+
+        let mut d_p = vec![0u8; size_p as usize];
+        let mut d_q = vec![0u8; size_q as usize];
+
+        let mut crt_coefficient = vec![0u8; size_p as usize];
+        let mut private_exponent = vec![0u8; self.get_size_of_modulus() as usize];
+
+        let mut primes_len = [
+            p.len() as symcrypt_sys::SIZE_T,
+            q.len() as symcrypt_sys::SIZE_T,
+        ];
+        let mut crt_lens = [
+            d_p.len() as symcrypt_sys::SIZE_T,
+            d_q.len() as symcrypt_sys::SIZE_T,
+        ];
+
+        unsafe {
+            // SAFETY: FFI calls
+            let result = symcrypt_sys::SymCryptRsakeyGetValue(
+                self.inner.0,
+                modulus_buffer.as_mut_ptr(),
+                modulus_buffer.len() as symcrypt_sys::SIZE_T,
+                pub_exp.as_mut_ptr(),
+                1 as symcrypt_sys::UINT32,
+                [p.as_mut_ptr(), q.as_mut_ptr()].as_mut_ptr(),
+                primes_len.as_mut_ptr(),
+                2 as symcrypt_sys::UINT32,
+                NumberFormat::MSB.to_symcrypt_format(),
+                self.key_usage.to_symcrypt_flag(),
+            );
+            if result != symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR {
+                return Err(result.into());
+            }
+
+            let result = symcrypt_sys::SymCryptRsakeyGetCrtValue(
+                self.inner.0,
+                [d_p.as_mut_ptr(), d_q.as_mut_ptr()].as_mut_ptr(),
+                crt_lens.as_mut_ptr(),
+                2,
+                crt_coefficient.as_mut_ptr(),
+                crt_coefficient.len() as symcrypt_sys::SIZE_T,
+                private_exponent.as_mut_ptr(),
+                private_exponent.len() as symcrypt_sys::SIZE_T,
+                NumberFormat::MSB.to_symcrypt_format(),
+                self.key_usage.to_symcrypt_flag(),
+            );
+            if result != symcrypt_sys::SYMCRYPT_ERROR_SYMCRYPT_NO_ERROR {
+                return Err(result.into());
+            }
+        }
+
+        let pub_exp_bytes = store_msb_first_u64(pub_exp[0], self.get_size_of_public_exponent())?;
```

Comment:
If you return the pub_exp as a u64 value, then, you will need to provide a public function to convert to multi-byte array and document.